pkg/certificate/manager.go

```diff
@@ -59,6 +55,9 @@ func (m *Manager) Start(checkInterval time.Duration, certRotation <-chan struct{
 }
 
 func (m *Manager) checkAndRotate() {
+	// NOTE: checkAndRotate can reintroduce a certificate that has been released, thereby creating an unbounded cache.
```

Comment:
Should we create an issue to track this so it can be revisted?

Reply:
I'm tempted not to, since I don't think it's very critical and happens in a few edge cases. 

For instance there's a ring buffer in the k8s informer  that is unbounded and a much larger concern for OOM's. My thought is this would get created then closed by the issue bot before it got prioritized for anybody to work on it. I'll leave it up to others though, if somebody else wants to create it!